File: 移除元素.cpp
#include<stdio.h>
int removeElement(int *nums, int numsSize, int val) {
    int i,k;
    for(i=0;i<numsSize;){
        if(val==nums[i]){
            for(k=i;k<numsSize-1;k++)
                nums[k]=nums[k+1];
            numsSize--;
        }
        else
            i++;
    }
    return numsSize;
}
int main(void){
	int i,len;
	int nums[10]={1,2,3,4,5,5,6,7,8,5};
	len=removeElement(nums,10,5);
	for(i=0;i<len;i++){
		printf("%3d",nums[i]);
	}
} 

File: 2020年11月/移除元素.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        c=0
        for i in range(len(nums)):
            if nums[i]!=val: #判断如果这个数不等于val，就从数组第一个数开始覆盖，之后返回c的大小
                nums[c]=nums[i]
                c+=1
        return c